packages/vote/net/server.go

```diff
@@ -71,12 +73,16 @@ func (vs *VoterServer) Run() error {
 		return err
 	}
 
+	vs.grpcServerMutex.Lock()
 	vs.grpcServer = grpc.NewServer()
+	vs.grpcServerMutex.Unlock()
```

Comment:
i am not 100% sure how grpc works, but I guess the `GracefulStop` only makes sense after a `Serve`. Thus, I would just defer the `Unlock` to also protect the register and serve.

Reply:
Actually what needs to be protected here is only the assignment, because it might happen that the Shutdown() is getting called by a different goroutine, while we are assigning the pointer.

Reply:
But what would happen now when both are called at the same time? `GracefulStop` could be completely executed and then we reach the `vs.grpcServer.Serve()` in `Run()`. Is then the server running although we called `Shutdown` at the same time?

Reply:
Ok, I've removed the initial goroutine, so that the shutdown gets called only after the assignment.